h/static/scripts/controllers/confirm-submit-controller.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+var Controller = require('../base/controller');
+
+/* Turn a normal submit element into one that shows a confirm dialog.
+ *
+ * The element's form will only be submitted if the user answers the
+ * confirmation dialog positively.
+ *
+ */
+class ConfirmSubmitController extends Controller {
+  constructor(element) {
+    super(element);
+
+    element.onclick = (event) => {
```

Comment:
The problem that you are having is rooted in how this controller handles the events specifically propagating your click interaction. There are two ways to fix this:

```
// option 1
element.onclick = (event) => {
  event.preventDefault();
  if (window.confirm(element.dataset.confirmMessage)) {
    element.onclick = null;
    setTimeout(()=>{
        element.click();
    }, 0);
  }
};
```

```
// option 2
element.addEventListener('click', (e)=>{
  if (!window.confirm(element.dataset.confirmMessage)) {
    e.preventDefault();
    e.stopPropagation();
    e.stopImmediatePropagation();
    return;
  }
}, /*capture*/true);
```

Option 1 works because it is pulling the click event into another control flow. Why that fixes this is not immediately obvious but I am not a huge fan of this approach to begin with (I have seen this breakdown on android as well)

Option 2 takes this an opposite approach of what you currently have. Your current implementation says "I will trigger the continue to happen if they accept confirmation" and the proposed solution says "I will trigger a shut down of the event if they do not confirm." So combined with use capture phase, we will trigger the confirmation and if they fail we do not let anything else hear about the event. I would go this route as it doesn't have to worry about any event bindings and triggering.


Side note: If you ever need to override a method like onclick or on{EventName} from an abstract helper, make sure you don't break any other possible overrides from other abstractions.
```
// something along these lines would not cause any conflicts with other overrides
// potentially sneaking in some painfully hard to find bugs
(()=>{
  var prevHandler = element.onclick;
  element.onclick = function(e){
    // .... logic here
    if(prevHandler){
      prevHandler(e);
    }
  }
})();
```